osmoutils/binary_search_test.go

```diff
@@ -33,8 +33,7 @@ func TestBinarySearch(t *testing.T) {
 		errTolerance  ErrTolerance
 		maxIterations int
 
-		expectedSolvedInput sdk.Int
```

Comment:
I like the idea of directly checking `targetOutput`, but I don't see why we need to remove `expectedSolvedInput` to ensure intermediate values are correct as well. I would personally prefer to keep both checks

Reply:
I thought there is no actual need for this intermediary check, because by checking that 
```
require.Equal(t, tc.errTolerance.CompareBigDec(output, tc.targetOutput), 0) // 0 - within errTolerance
```
outputs match, we can imply that an actualSolvedInput is correct indeed, because an output's value depends on it
```
output, err := tc.f(actualSolvedInput)
```
I might be wrong about this and if that is the case I will return an expectedSolvedInput back